packages/react/src/components/OverflowSet/OverflowButton.tsx

```diff
@@ -4,26 +4,51 @@ import { IKeytipProps } from '../../Keytip';
 import { IOverflowSetItemProps, IOverflowSetProps } from './OverflowSet.types';
 import { useConst } from '@fluentui/react-hooks';
 
+const unregisteredKeytipID = 'keytipToBeRegistered';
+
+const keytipHasBeenRegistered = (id: string) => {
+  return id.indexOf(unregisteredKeytipID) === -1;
+};
+
+const registerPersistedKeytip = (
+  keytip: IKeytipProps,
+  keytipManager: KeytipManager,
+  persistedKeytips: { [uniqueID: string]: IKeytipProps },
+  oldKeytipId: string,
+) => {
+  const uniqueID = keytipManager.register(keytip, true);
+  // Update map
+  persistedKeytips[uniqueID] = keytip;
+  delete persistedKeytips[oldKeytipId];
+};
+
 const useKeytipRegistrations = (
   persistedKeytips: { [uniqueID: string]: IKeytipProps },
   keytipManager: KeytipManager,
 ) => {
-  React.useEffect(() => {
+  React.useLayoutEffect(() => {
```

Comment:
Looks like this fn registers/updates a keytip object, is there a reason why this was changed to useLayoutEffect? If it's because of the update case, should these be split and only the update uses useLayoutEffect?

Reply:
I'm following a similar pattern as we have in useKeytipData

The old code used to run on didmount/unmount/update etc. On mount/update I want to register/update, on unmount I want to unregister keytips

Reply:
but does it need to run before paint? (Usually, you only want to use `useLayoutEffect` if it causes a flash in the UI (e.g. another render)

Reply:
Removed usage of useLayoutEffect

Reply:
@ecraig12345 could you take another look at this when you get the chance?